polyfill/lib/timezone.mjs

```diff
@@ -1,115 +1,148 @@
-  const before = ES.GetTimeZoneOffsetNanoSeconds(utcns - 86400000000000n, GetSlot(this, IDENTIFIER));
-  const after = ES.GetTimeZoneOffsetNanoSeconds(utcns + 86400000000000n, GetSlot(this, IDENTIFIER));
-  const diff = ES.CastToDuration({
-    nanoseconds: Number(after - before)
-  });
-  switch (disambiguation) {
-    case 'earlier':
-      const earlier = dateTime.minus(diff);
-      return this.getAbsoluteFor(earlier, disambiguation);
-    case 'later':
-      const later = dateTime.plus(diff);
-      return this.getAbsoluteFor(later, disambiguation);
-    default:
-      throw new RangeError(`no such absolute found`);
+  getTransitions(startingPoint) {
+    startingPoint = ES.GetIntrinsic('%Temporal.absolute%')(startingPoint);
+    let { ms } = GetSlot(startingPoint, EPOCHNANOSECONDS);
+    const Absolute = ES.GetIntrinsic('%Temporal.Absolute%');
+    return {
+      next: () => {
+        ms = ES.GetTimeZoneNextTransition(ms, GetSlot(this, IDENTIFIER));
+        const done = epochNanoseconds !== null;
+        const value = epochNanoseconds !== null ? null : new Absolute(epochNanoseconds);
+        return { done, value };
+      }
+    };
   }
-};
-TimeZone.prototype.getTransitions = function getTransitions(startingPoint) {
-  let epochNanoseconds = startingPoint.getEpochNanoseconds();
-  const Absolute = ES.GetIntrinsic('%Temporal.Absolute%');
-  return {
-    next: () => {
-      epochNanoseconds = ES.GetTimeZoneNextTransition(epochNanoseconds, GetSlot(this, IDENTIFIER));
-      const done = epochNanoseconds !== null;
-      const value = epochNanoseconds !== null ? null : new Absolute(epochNanoseconds);
-      return { done, value };
-    }
-  };
-};
-TimeZone.prototype.toString = TimeZone.prototype.toJSON = function toString() {
-  return this.name;
-};
-TimeZone.fromString = function fromString(isoString) {
-  isoString = ES.ToString(isoString);
-  const match = TZRE.exec(isoString);
-  if (!match) throw new RangeError(`invalid timezone string: ${isoString}`);
-  return new TimeZone(match[1]);
-};
-TimeZone[Symbol.iterator] = function() {
-  const iter = ZONES[Symbol.iterator]();
-  return {
-    next: () => {
-      while (true) {
-        let { value, done } = iter.next();
-        if (done) return { done };
-        try {
-          value = TimeZone(value);
-          done = false;
-          return { done, value };
-        } catch (ex) {}
+  toString() {
+    return this.name;
+  }
+  static fromString(isoString) {
+    isoString = ES.ToString(isoString);
+    const match = STRING.exec(isoString);
+    if (!match) throw new RangeError(`invalid timezone: ${isoString}`);
+    const zone = match[1] ? 'UTC' : match[3] || match[2];
+    return new TimeZone(zone);
+  }
+  static from(...args) {
+    return ES.GetIntrinsic('%Temporal.timezone%')(...args);
+  }
+}
+if ('undefined' !== typeof Symbol) {
+  TimeZone[Symbol.iterator] = function() {
```

Comment:
I'm not so sure if we should have an API to iterate over all timezones, as opposed to just a way to check whether a timezone is valid (as Intl currently does). I'd imagine this API alongside other requested APIs to enumerate supported locales, etc. What was the motivation?

Reply:
The reason was that this was one of the frequently requested things. It's incredibly useful for dropdown lists for choosing a time-zone. That being said, I'm not insisting on it. As a matter of fact I was arguing against it and was persuaded by (if I recall correctly) you, to have this (albeit as a get function to an array which seemed worse than an iterator)